### ISSUE TITLE
Add scheduling functions to the template

### ISSUE THREAD

**Issue by yipliu** [ashleve/lightning-hydra-template#433]
Hi,

Almost all models now consider the use of **scheduling**, however, this feature has not been added to the latest templates. 

I have considered adding, but the method used is not perfect, looking forward to your solution.

The following method is the solution I am using
```python
    def configure_optimizers(self):
        
        optimizer = torch.optim.Adam(
            params=self.parameters(), lr=self.hparams.lr, weight_decay=self.hparams.weight_decay
        )

        scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
            optimizer=optimizer,
            mode=self.hparams.mode,
            patience=self.hparams.patience,
            factor=self.hparams.factor,
            min_lr=self.hparams.min_lr,
        )

        return {
            "optimizer": optimizer,
            "lr_scheduler": {
                "scheduler": scheduler,
                "monitor": "val/loss",  
                "interval": "epoch",
                "frequency": 1,
            },
        }
```
I have to point out that my implementation is very complicated.

The following method was conceived by me, but not verified
```python
optimizer = self.hparams.optimizer(params=self.parameters())
        return {
            "optimizer": optimizer ,
            "lr_scheduler":{
                "scheduler": self.hparams.scheduler(optimizer=optimizer ,lr_lambda=lr_lambda)
            }
        }
```

```yaml
optimizer:
  _target_: torch.optim.Adam
  _partial_: true  
  lr: 0.005
  betas: (0.9, 0.98) 
  eps: 1e-9

scheduler:
  _target_: torch.optim.lr_scheduler.LambdaLR
  _partial_: true
  last_epoch: -1
```

If this feature needs to be added, and the code logic symbol template, I'd be happy to start experimenting with it.

**Comment by ashleve** [comment]
Sure, we can add the scheduler example. I think the second approach with partial initialization should work. You can go for the PR if you'd like to!
